packages/eslint-plugin/src/rules/prefer-promise-reject-errors.ts

```diff
@@ -0,0 +1,147 @@
+      },
+    ],
+    messages: {
+      rejectAnError: 'Expected the Promise rejection reason to be an Error.',
+    },
+  },
+  defaultOptions: [
+    {
+      allowEmptyReject: false,
+    },
+  ],
+  create(context, [options]) {
+    const services = getParserServices(context);
+    const program = services.program;
+
+    function checkRejectCall(callExpression: TSESTree.CallExpression): void {
+      const argument = callExpression.arguments.at(0);
+      if (!argument && options.allowEmptyReject) {
+        return;
+      }
+
+      if (
+        !argument ||
+        !isErrorLike(program, services.getTypeAtLocation(argument))
+      ) {
+        context.report({
+          node: callExpression,
+          messageId: 'rejectAnError',
+        });
+      }
+    }
+
+    function isPromiseConstructorLike(type: ts.Type): boolean {
+      const symbol = type.getSymbol();
+
+      if (symbol?.getName() === 'PromiseConstructor') {
+        const declarations = symbol.getDeclarations() ?? [];
+        for (const declaration of declarations) {
+          const sourceFile = declaration.getSourceFile();
+          if (program.isSourceFileDefaultLibrary(sourceFile)) {
+            return true;
+          }
+        }
+      }
+
+      return false;
+    }
+
+    function skipChainExpression<T>(node: T): T & TSESTree.ChainElement {
+      // @ts-expect-error https://github.com/typescript-eslint/typescript-eslint/issues/8008
+      return node.type === AST_NODE_TYPES.ChainExpression
+        ? (node as TSESTree.ChainExpression).expression
+        : node;
+    }
+
+    return {
+      CallExpression(node): void {
+        const callee = skipChainExpression(node.callee);
+        if (
+          callee.type !== AST_NODE_TYPES.MemberExpression ||
+          (callee.computed
+            ? callee.property.type === AST_NODE_TYPES.Literal &&
+              callee.property.value !== 'reject'
+            : callee.property.name !== 'reject') ||
+          !isPromiseConstructorLike(services.getTypeAtLocation(callee.object))
+        ) {
+          return;
+        }
+
+        checkRejectCall(node);
+      },
+      NewExpression(node): void {
+        const callee = skipChainExpression(node.callee);
+        if (!isPromiseConstructorLike(services.getTypeAtLocation(callee))) {
+          return;
+        }
+
+        const executor = node.arguments.at(0);
+        if (!executor || !isFunction(executor)) {
+          return;
+        }
+        const rejectParamNode = executor.params.at(1);
+        if (!rejectParamNode || !isIdentifier(rejectParamNode)) {
+          return;
+        }
+
+        const rejectVariable = getDeclaredVariables(context, executor).find(
+          variable => variable.identifiers.includes(rejectParamNode),
+        );
+        /* istanbul ignore if */ if (!rejectVariable) {
+          return;
+        }
+
+        rejectVariable.references.forEach(ref => {
+          if (
+            !ref.isRead() ||
```

Comment:
[Testing] I tried removing `!ref.isRead() ||` and none of the unit tests failed. Is there a reason to include it that can be tested?

Reply:
You're right, thanks, check for `CallExpression` is already a check for `isRead`